pkg/master/master_test.go

```diff
@@ -370,114 +393,109 @@ func TestAPIVersionOfDiscoveryEndpoints(t *testing.T) {
 
-		assert.Equal(expectPreferredVersion[0], groupList.Groups[0].PreferredVersion)
-		assert.Equal(expectPreferredVersion[1], groupList.Groups[1].PreferredVersion)
+	assert.Equal(expectPreferredVersion[0], groupList.Groups[0].PreferredVersion)
+	assert.Equal(expectPreferredVersion[1], groupList.Groups[1].PreferredVersion)
 
-		thirdPartyGV := unversioned.GroupVersionForDiscovery{GroupVersion: "company.com/v1", Version: "v1"}
-		master.addThirdPartyResourceStorage("/apis/company.com/v1", nil,
-			unversioned.APIGroup{
-				Name:             "company.com",
-				Versions:         []unversioned.GroupVersionForDiscovery{thirdPartyGV},
-				PreferredVersion: thirdPartyGV,
-			})
+	thirdPartyGV := unversioned.GroupVersionForDiscovery{GroupVersion: "company.com/v1", Version: "v1"}
+	master.addThirdPartyResourceStorage("/apis/company.com/v1", nil,
+		unversioned.APIGroup{
+			Name:             "company.com",
+			Versions:         []unversioned.GroupVersionForDiscovery{thirdPartyGV},
+			PreferredVersion: thirdPartyGV,
+		})
 
-		resp, err = http.Get(server.URL + "/apis")
-		if !assert.NoError(err) {
-			t.Errorf("unexpected error: %v", err)
-		}
+	resp, err = http.Get(server.URL + "/apis")
+	if !assert.NoError(err) {
+		t.Errorf("unexpected error: %v", err)
+	}
 
-		assert.Equal(http.StatusOK, resp.StatusCode)
+	assert.Equal(http.StatusOK, resp.StatusCode)
 
-		assert.NoError(decodeResponse(resp, &groupList))
-		if err != nil {
-			t.Fatalf("unexpected error: %v", err)
-		}
+	assert.NoError(decodeResponse(resp, &groupList))
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
 
-		thirdPartyGroupName := "company.com"
-		thirdPartyExpectVersions := []unversioned.GroupVersionForDiscovery{thirdPartyGV}
-
-		assert.Equal(4, len(groupList.Groups))
-		// autoscaling group
-		assert.Equal(expectGroupNames[0], groupList.Groups[0].Name)
-		assert.Equal(expectVersions[0], groupList.Groups[0].Versions)
-		assert.Equal(expectPreferredVersion[0], groupList.Groups[0].PreferredVersion)
-		// batch group
-		assert.Equal(expectGroupNames[1], groupList.Groups[1].Name)
-		assert.Equal(expectVersions[1], groupList.Groups[1].Versions)
-		assert.Equal(expectPreferredVersion[1], groupList.Groups[1].PreferredVersion)
-		// third party
-		assert.Equal(thirdPartyGroupName, groupList.Groups[2].Name)
-		assert.Equal(thirdPartyExpectVersions, groupList.Groups[2].Versions)
-		assert.Equal(thirdPartyGV, groupList.Groups[2].PreferredVersion)
-		// extensions group
-		assert.Equal(expectGroupNames[2], groupList.Groups[3].Name)
-		assert.Equal(expectVersions[2], groupList.Groups[3].Versions)
-		assert.Equal(expectPreferredVersion[2], groupList.Groups[3].PreferredVersion)
-	*/
+	thirdPartyGroupName := "company.com"
+	thirdPartyExpectVersions := []unversioned.GroupVersionForDiscovery{thirdPartyGV}
+
+	assert.Equal(4, len(groupList.Groups))
+	// autoscaling group
+	assert.Equal(expectGroupNames[0], groupList.Groups[0].Name)
```

Comment:
Now it uses map instead of lists, with group name as map key, so there is no more hard-coding. PTAL. Thanks.